grakn-graql/src/main/java/ai/grakn/graql/internal/hal/HALConceptRepresentationBuilder.java

```diff
@@ -176,4 +179,36 @@ private static void attachSingleGeneratedRelation(Representation currentHal, Con
         });
         return roleTypes;
     }
+
+    static Schema.BaseType getBaseType(Instance instance) {
```

Comment:
Maybe it's worth making a ConceptAdmin API so we get things like the BaseType and vertex more easily ? Not sure how valuable that would be.